staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -140,7 +158,13 @@ func (s *Validator) validateExpressions(fldPath *field.Path, sts *schema.Structu
 			errs = append(errs, field.InternalError(fldPath, fmt.Errorf("oldSelf validation not implemented")))
 			continue // todo: wire oldObj parameter
 		}
-		evalResult, _, err := compiled.Program.Eval(activation)
+		evalResult, evalDetails, err := compiled.Program.Eval(activation)
```

Comment:
Do we want to check specifically for the runtime cost limit exceeded error here and output a more descriptive error, maybe with details about the cost used and how it relates to the totals?

Reply:
Added check